storage/client/src/test/java/com/azure/storage/queue/QueueClientTests.java

```diff
@@ -0,0 +1,651 @@
+
+        AccessPolicy accessPolicy = new AccessPolicy()
+            .permission("r")
+            .start(OffsetDateTime.of(LocalDateTime.of(2000, 1, 1, 0, 0), ZoneOffset.UTC))
+            .expiry(OffsetDateTime.of(LocalDateTime.of(2020, 1, 1, 0, 0), ZoneOffset.UTC));
+
+        for (int i = 0; i < 6; i++) {
+            permissions.add(new SignedIdentifier()
+                .id("policy" + i)
+                .accessPolicy(accessPolicy));
+        }
+
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        try {
+            client.setAccessPolicy(permissions);
+            fail("Attempting to set more than five access policies on a queue that doesn't exist should throw an exception");
+        } catch (Exception exception) {
+            TestHelpers.assertExceptionStatusCode(exception, 400);
+        }
+    }
+
+    @Override
+    public void enqueueMessage() {
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        String messageText = "test message";
+        TestHelpers.assertResponseStatusCode(client.enqueueMessage(messageText), 201);
+
+        Iterator<PeekedMessage> response = client.peekMessages().iterator();
+        assertEquals(messageText, response.next().messageText());
+        assertFalse(response.hasNext());
+    }
+
+    @Override
+    public void enqueueEmptyMessage() {
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        String messageText = "";
+        TestHelpers.assertResponseStatusCode(client.enqueueMessage(messageText), 201);
+
+        Iterator<PeekedMessage> response = client.peekMessages().iterator();
+        assertNull(response.next().messageText());
+        assertFalse(response.hasNext());
+    }
+
+    @Override
+    public void enqueueShortTimeToLiveMessage() {
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        String messageText = "test message";
+        TestHelpers.assertResponseStatusCode(client.enqueueMessage(messageText, Duration.ofSeconds(0), Duration.ofSeconds(2)), 201);
+
+        TestHelpers.sleep(Duration.ofSeconds(5));
+        Iterator<PeekedMessage> response = client.peekMessages().iterator();
+        assertFalse(response.hasNext());
+    }
+
+    @Override
+    public void enqueueQueueDoesNotExist() {
+        try {
+            client.enqueueMessage("This should fail");
+            fail("Attempting to enqueue a message on a queue that doesn't exist should throw an exception");
+        } catch (Exception exception) {
+            TestHelpers.assertExceptionStatusCode(exception, 404);
+        }
+    }
+
+    @Override
+    public void dequeueMessage() {
+        TestHelpers.assertResponseStatusCode(client.create(), 201);
+
+        String messageText = "test message";
+        TestHelpers.assertResponseStatusCode(client.enqueueMessage(messageText), 201);
+
+        Iterator<DequeuedMessage> response = client.dequeueMessages().iterator();
```

Comment:
Did we reach any conclusion if we want to merge DequedMessage and PeekedMessage?

Reply:
I'll follow-up with Ted about this.